kedro/pipeline/node.py

```diff
@@ -511,6 +512,83 @@ def _process_inputs_for_bind(inputs: None | str | list[str] | dict[str, str]):
             kwargs = inputs
         return args, kwargs
 
+    def to_dict(self) -> NodeAsDict:
+        """Create a dictionary representation of the node suitable for dumping to JSON.
+
+        To ensure this representation can be loaded back in the same environment using
+        :py:meth:`kedro.pipeline.node.Node.from_dict`, the following constraints apply
+        to the node function:
+
+        - It cannot be a lambda function.
+        - It cannot be an inner function.
+        - It cannot be defined in ``__main__`` module.
+        """
+        err_msg = f"Cannot convert node '{self}' to dict."
+        qualname = self._func.__qualname__
+        if qualname == "<lambda>":
```

Comment:
don't know if this is safer
https://stackoverflow.com/a/24578359/2010808

I'd also be interested to know if we could enable an `unsafe` mode and include `inspect.getsource` as a fallback for this set of of `ValueError`.

Reply:
https://www.tutorialspoint.com/How-to-retrieve-source-code-from-Python-objects#:~:text=to%20the%20console.-,The%20inspect.,present%20in%20the%20source%20code.

Reply:
I guess we can even `pickle.dumps()` in `unsafe` mode, which might unlock some additional cases compared to `inspect.getsource()`.